TorqueDataConnect/resources/js/TorqueDataConnectEdit.js

```diff
@@ -0,0 +1,224 @@
+/*@nomin*/
+function addEditButtonListeners () {
+    $(".torque-edit-button.paragraph").click(handleParagraphEdit);
+    $(".torque-edit-button.list").click(handleListEdit);
+
+    const [sheetName, , key] = $("#page-info")
+        .data("location")
+        .replace(".mwiki", "")
+        .split("/");
+    window.sheetName = sheetName;
+    window.key = key;
+}
+
+$(document).ready(() => {
+    addEditButtonListeners();
+});
+
+async function submitEdit (field, value, callback) {
+    const postData = {
+        newValues: JSON.stringify({ [field]: value }),
+        sheetName: window.sheetName,
+        key: window.key,
+        title: mw.config.values.wgTitle,
+    };
+    try {
+        // Gets CORS token for POST
+        const corsResult = await $.ajax({
+            type: "GET",
+            url: "/lfc/api.php?action=query&format=json&meta=tokens",
+            dataType: "json",
+        });
+
+        const token = encodeURIComponent(corsResult.query.tokens.csrftoken);
+        const actionName = "torquedataconnectsubmitedit";
+        const results = await $.ajax({
+            type: "POST",
+            url: `${mw.config.values.wgScriptPath}/api.php?action=${actionName}&format=json&token=${token}`,
+            data: postData,
+            dataType: "json",
+        });
+        $('.torque-wrapper').replaceWith(results.html);
+        addEditButtonListeners();
+    } catch (error) {
+        console.error(error);
+    }
+}
+
+const textArea = (v) => $(`<textarea name="" type="text">${v}</textarea>`);
+// Returns a jquery cancel and save button side-by-side
+const saveButtons = (fieldName, originalValue, onCancel, onSave) => {
+    const cancelBtn = $('<span class="torque-save-cancel">Cancel</span>');
```

Comment:
Hm.  Is there a way to have "Cancel" come from the localization files?  I know that there is some infrastructure in mediawiki to get those messages into javascript, but I'm not sure how those match with what you're doing.  "Save" below runs into the same thing.  We can also put these in the template or CSS, which would allow other users of the extension to internationalize/configure it as they like.

Reply:
I think this is a great idea but should it block shipping the feature? None of our users use localization, right?

Reply:
Then we should create an issue to later internationalize the javascript code when this goes live.  As for our users, we have had cases where we've used localization in order to create configuration.  Like, for instance, they would prefer to use "change" instead of "edit."  We can make that update in the template, but the javascript can't follow along without a code patch.